lib/libtsunami/src/drawables/form.cpp

```diff
@@ -52,74 +52,99 @@ Form::Form(int x, int y, uint width, uint height, bool is_popup, int z_index, bo
 }
 
 Form::~Form() {
+	clearObjects();
+	clearBottomObjects();
+
+	if (m_columns_attributes) {
+		free(m_columns_attributes);
+		m_columns_attributes = nullptr;
+	}
+
+	if (m_rows_attributes) {
+		free(m_rows_attributes);
+		m_rows_attributes = nullptr;
+	}
+
 	if (m_cursor_animation != nullptr) {
 		m_cursor_animation->complete(m_cursor);
 		delete m_cursor_animation;
 		m_cursor_animation = nullptr;
 	}
 
-	if (m_cursor != nullptr) {
-		subRemove(m_cursor);
-		delete m_cursor;
-		m_cursor = nullptr;
-	}
-
 	if (m_bottom_cursor_animation != nullptr) {
 		m_bottom_cursor_animation->complete(m_bottom_cursor);
 		delete m_bottom_cursor_animation;
 		m_bottom_cursor_animation = nullptr;
 	}
 
-	if (m_bottom_cursor != nullptr) {
-		subRemove(m_bottom_cursor);
-		delete m_bottom_cursor;
-		m_bottom_cursor = nullptr;
+	if (m_cursor != nullptr) {
+		m_cursor->setFinished();
 	}
 
-	clearObjects();
-	clearBottomObjects();
+	if (m_bottom_cursor != nullptr) {
+		m_bottom_cursor->setFinished();
+	}	
 
-	if (m_columns_attributes) {
-		free(m_columns_attributes);
-		m_columns_attributes = nullptr;
+	if (m_background_rectangle != nullptr) {
+		m_background_rectangle->setFinished();
+	}
+	
+	if (m_title_rectangle != nullptr) {
+		m_title_rectangle->setFinished();
+	}
+	
+	if (m_body_rectangle != nullptr) {
+		m_body_rectangle->setFinished();
+	}
+	
+	if (m_bottom_rectangle != nullptr) {
+		m_bottom_rectangle->setFinished();
+	}
+	
+	if (m_title_label != nullptr) {
+		m_title_label->setFinished();
 	}
 
-	if (m_rows_attributes) {
-		free(m_rows_attributes);
-		m_rows_attributes = nullptr;
+	this->setFinished();
+	this->subRemoveFinished();
+	thd_sleep(100);
```

Comment:
Why you need it?

Reply:
It happened to me that because there were many objects, when I destroyed them I got an error, when I put it to sleep it stopped doing so, I did several tests for that.

Reply:
Do you think it affects anything?

Reply:
I think this is not a fix, but just a deeply hidden bug :)

Reply:
Yes, it may be, I haven't detected it yet, I can remove it in the meantime :D

Reply:
In any case I would change it to thd_pass, it is the one I have been using for this and then I will eliminate it over time.

Reply:
Block the video thread, I don't know, but I'll try that later.

Reply:
Well, when you find it, then delete it. Just make a note like FIXME. And if it works with thd_pass better use it.

Reply:
I like it, I'll do that, I'll put thd_pass but with the FIXME comment.

Reply:
Ready, now I only have to fix errors, I no longer plan to add features to this version :)

Reply:
I'm ready to see the merge.